pkgs/data/misc/iana-etc/default.nix

```diff
@@ -1,17 +1,81 @@
-{stdenv, fetchurl}:
+{ stdenv, fetchurl }:
 
-stdenv.mkDerivation rec {
-  name = "iana-etc-2.30";
+let
 
-  src = fetchurl {
-    url = "http://sethwklein.net/${name}.tar.bz2";
-    sha256 = "03gjlg5zlwsdk6qyw3v85l129rna5bpm4m7pzrp864h0n97qg9mr";
+  iana = {
+    
+    protocolsXML = fetchurl {
+      urls = [
+        "https://www.iana.org/assignments/protocol-numbers/protocol-numbers.xml"
+        "ftp://ftp.iana.org/assignments/protocol-numbers/protocol-numbers.xml"
+      ];
```

Comment:
That was also my concern when writing this because IANA does not version those files (contrary to `tzdata` for example). So if/when the source is edited this definition will need an update to update the SHA hashes.

But for me it is acceptable since :
1. `protocols` rarely change, `services` is modified more often but not every two days ;
2. we should have the package substitute available in the cache so it won't break networking on NixOS machines.

For illustration : the source of the old `iana-etc` package was missing for years (wrong URL) and no one noticed until #23167 

But it is totally possible to change that. The alternative approach I envisioned is simply store a copy of the two XML right here on our GH repo next to `default.nix` and just `import` them.